node/src/protocol.rs

```diff
@@ -137,12 +137,12 @@ impl Payload for Message {
         }
     }
 
-    fn is_unsafe_for_joiners(&self) -> bool {
+    fn is_unsafe_for_syncing_peers(&self) -> bool {
         match self {
             Message::Consensus(_) => false,
             Message::DeployGossiper(_) => false,
             Message::AddressGossiper(_) => false,
-            // Trie requests can deadlock between joiners.
+            // Trie requests can deadlock between syncing nodes.
             Message::GetRequest { tag, .. } if *tag == Tag::TrieOrChunk => true,
```

Comment:
We have a trait describing the same information now in the chain synchronizer, which we need to clean up (`CanUseJoiners`).

I think it'd be best to put the `CanUseJoiners` functionality onto the `Tag` for now:

```rust
impl Tag {
    fn can_use_joiners(self) -> bool;
}
```

For any function handling an item `T: Item`, we can then use `(<T as Item>::Tag).is_unsafe_for_joiners()`.

Reply:
I'd rather postpone this change to another PR to avoid scope creep as neither the trait nor the functions was introduced as part of this PR.

Reply:
Fair enough!